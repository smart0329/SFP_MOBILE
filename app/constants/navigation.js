// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import keyMirror from 'mattermost-redux/utils/key_mirror';

const NavigationTypes = keyMirror({
    NAVIGATION_RESET: null,
    NAVIGATION_CLOSE_MODAL: null,
    NAVIGATION_NO_TEAMS: null,
    RESTART_APP: null,
    NAVIGATION_ERROR_TEAMS: null,
    NAVIGATION_SHOW_OVERLAY: null,
    CLOSE_MAIN_SIDEBAR: null,
    MAIN_SIDEBAR_DID_CLOSE: null,
    MAIN_SIDEBAR_DID_OPEN: null,
    CLOSE_SETTINGS_SIDEBAR: null,
    BLUR_POST_TEXTBOX: null,
});

export default NavigationTypes;
