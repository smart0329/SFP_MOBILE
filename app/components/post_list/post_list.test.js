// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import * as NavigationActions from 'app/actions/navigation';
import PostList from './post_list';

jest.useFakeTimers();
jest.mock('react-intl');

describe('PostList', () => {
    const serverURL = 'https://server-url.fake';
    const deeplinkRoot = 'mattermost://server-url.fake';

    const baseProps = {
        actions: {
            handleSelectChannelByName: jest.fn(),
            loadChannelsByTeamName: jest.fn(),
            refreshChannelWithRetry: jest.fn(),
            selectFocusedPostId: jest.fn(),
            setDeepLinkURL: jest.fn(),
        },
        channelId: 'channel-id',
        deepLinkURL: '',
        lastPostIndex: -1,
        postIds: ['post-id-1', 'post-id-2'],
        serverURL,
        siteURL: 'https://site-url.fake',
        theme: Preferences.THEMES.default,
    };

    const deepLinks = {
        permalink: deeplinkRoot + '/team-name/pl/pl-id',
        channel: deeplinkRoot + '/team-name/channels/channel-name',
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <PostList {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('setting permalink deep link', () => {
        const showModalOverCurrentContext = jest.spyOn(NavigationActions, 'showModalOverCurrentContext');
        const wrapper = shallow(
            <PostList {...baseProps}/>,
        );

        wrapper.setProps({deepLinkURL: deepLinks.permalink});
        expect(baseProps.actions.setDeepLinkURL).toHaveBeenCalled();
        expect(baseProps.actions.selectFocusedPostId).toHaveBeenCalled();
        expect(showModalOverCurrentContext).toHaveBeenCalled();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('setting channel deep link', () => {
        const wrapper = shallow(
            <PostList {...baseProps}/>,
        );

        wrapper.setProps({deepLinkURL: deepLinks.channel});
        expect(baseProps.actions.setDeepLinkURL).toHaveBeenCalled();
        expect(baseProps.actions.handleSelectChannelByName).toHaveBeenCalled();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call flatListScrollToIndex only when ref is set and index is in range', () => {
        jest.spyOn(global, 'requestAnimationFrame').mockImplementation((cb) => cb());

        const wrapper = shallow(
            <PostList {...baseProps}/>,
        );
        const instance = wrapper.instance();
        const flatListScrollToIndex = jest.spyOn(instance, 'flatListScrollToIndex');
        const indexInRange = baseProps.postIds.length;
        const indexOutOfRange = [-1, indexInRange + 1];

        instance.flatListRef = {
            current: null,
        };
        instance.scrollToIndex(indexInRange);
        expect(flatListScrollToIndex).not.toHaveBeenCalled();

        instance.flatListRef = {
            current: {
                scrollToIndex: jest.fn(),
            },
        };
        for (const index of indexOutOfRange) {
            instance.scrollToIndex(index);
            expect(flatListScrollToIndex).not.toHaveBeenCalled();
        }

        instance.scrollToIndex(indexInRange);
        expect(flatListScrollToIndex).toHaveBeenCalled();
    });

    test('should load more posts if available space on the screen', () => {
        const wrapper = shallow(
            <PostList {...baseProps}/>,
        );
        const instance = wrapper.instance();
        instance.loadToFillContent = jest.fn();

        wrapper.setProps({
            extraData: false,
        });
        expect(instance.loadToFillContent).toHaveBeenCalledTimes(0);

        wrapper.setState({
            postListHeight: 500,
            contentHeight: 200,
        });
        expect(instance.loadToFillContent).toHaveBeenCalledTimes(1);

        wrapper.setProps({
            extraData: true,
        });

        expect(instance.loadToFillContent).toHaveBeenCalledTimes(1);
    });
});
