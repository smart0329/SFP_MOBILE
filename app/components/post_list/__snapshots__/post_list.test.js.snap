// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList setting channel deep link 1`] = `
<FlatList
  ListFooterComponent={[Function]}
  contentContainerStyle={
    Object {
      "paddingTop": 5,
    }
  }
  data={
    Array [
      "post-id-1",
      "post-id-2",
    ]
  }
  disableVirtualization={false}
  extraData={
    Array [
      "channel-id",
      undefined,
      undefined,
      undefined,
    ]
  }
  horizontal={false}
  initialNumToRender={7}
  inverted={true}
  keyExtractor={[Function]}
  keyboardDismissMode="interactive"
  keyboardShouldPersistTaps="handled"
  maintainVisibleContentPosition={
    Object {
      "autoscrollToTopThreshold": 60,
      "minIndexForVisible": 0,
    }
  }
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollToIndexFailed={[Function]}
  refreshControl={
    <RefreshControl
      colors={
        Array [
          "#3d3c40",
        ]
      }
      onRefresh={[Function]}
      refreshing={false}
      tintColor="#3d3c40"
    />
  }
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={60}
  style={
    Object {
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  windowSize={50}
/>
`;

exports[`PostList setting permalink deep link 1`] = `
<FlatList
  ListFooterComponent={[Function]}
  contentContainerStyle={
    Object {
      "paddingTop": 5,
    }
  }
  data={
    Array [
      "post-id-1",
      "post-id-2",
    ]
  }
  disableVirtualization={false}
  extraData={
    Array [
      "channel-id",
      undefined,
      undefined,
      undefined,
    ]
  }
  horizontal={false}
  initialNumToRender={7}
  inverted={true}
  keyExtractor={[Function]}
  keyboardDismissMode="interactive"
  keyboardShouldPersistTaps="handled"
  maintainVisibleContentPosition={
    Object {
      "autoscrollToTopThreshold": 60,
      "minIndexForVisible": 0,
    }
  }
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollToIndexFailed={[Function]}
  refreshControl={
    <RefreshControl
      colors={
        Array [
          "#3d3c40",
        ]
      }
      onRefresh={[Function]}
      refreshing={false}
      tintColor="#3d3c40"
    />
  }
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={60}
  style={
    Object {
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  windowSize={50}
/>
`;

exports[`PostList should match snapshot 1`] = `
<FlatList
  ListFooterComponent={[Function]}
  contentContainerStyle={
    Object {
      "paddingTop": 5,
    }
  }
  data={
    Array [
      "post-id-1",
      "post-id-2",
    ]
  }
  disableVirtualization={false}
  extraData={
    Array [
      "channel-id",
      undefined,
      undefined,
      undefined,
    ]
  }
  horizontal={false}
  initialNumToRender={7}
  inverted={true}
  keyExtractor={[Function]}
  keyboardDismissMode="interactive"
  keyboardShouldPersistTaps="handled"
  maintainVisibleContentPosition={
    Object {
      "autoscrollToTopThreshold": 60,
      "minIndexForVisible": 0,
    }
  }
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollToIndexFailed={[Function]}
  refreshControl={
    <RefreshControl
      colors={
        Array [
          "#3d3c40",
        ]
      }
      onRefresh={[Function]}
      refreshing={false}
      tintColor="#3d3c40"
    />
  }
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={60}
  style={
    Object {
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  windowSize={50}
/>
`;
